src/main/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProvider.java

```diff
@@ -0,0 +1,398 @@
+    private static final String SCONTROL_HOSTS = "hostList";
+    private static final String SCONTROL_PARTITION_NAME = "partitionName";
+    private static final String STANDARD_SLURM_DELIMETER = "\\|";
+    private static final String LEFT_SQUARE_BRACKET_SIGN = "\\[";
+    private static final String HYPHEN_SIGN = "-";
+
+    private static final Pattern NODES_RANGE_REGEX = Pattern.compile("[a-zA-Z]+\\[\\d+-\\d+]");
+    private static final int SCONTROL_OUPUT_SIZE = 42;
+    private static final int SCONTROL_OUPUT_PARTNAME_INDEX = 33;
+    private static final int SCONTROL_OUPUT_NODES_INDEX = 31;
+    private static final int SCONTROL_OUPUT_USERGROUPS_INDEX = 6;
+    private static final int SCONTROL_OUPUT_CPUS_INDEX = 2;
+
+    private final SimpleCmdExecutor simpleCmdExecutor;
+
+    /**
+     * An object that forms the structure of an executable command according to a template.
+     */
+    private final GridEngineCommandCompiler commandCompiler;
+
+    /**
+     * Returns Slurm Grid Engine.
+     *
+     * @return current engine type - Slurm Grid Engine
+     */
+    @Override
+    public CommandType getProviderType() {
+        return CommandType.SLURM;
+    }
+
+    @Override
+    public Queue registerQueue(final QueueVO registrationRequest) {
+        checkRegistrationRequest(registrationRequest);
+        final Context context = prepareContext(registrationRequest);
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SCONTROL_COMMAND, context));
+        checkIsResultIsCorrect(result);
+        return Queue.builder()
+                .name(registrationRequest.getName())
+                .hostList(registrationRequest.getHostList())
+                .allowedUserGroups(registrationRequest.getAllowedUserGroups())
+                .slots(new SlotsDescription(2, Collections.emptyMap()))
+                .build();
+    }
+
+    @Override
+    public Queue updateQueue(final QueueVO updateRequest) {
+        checkRegistrationRequest(updateRequest);
+
+        final String updateName = updateRequest.getName();
+        final List<String> updateHostList = updateRequest.getHostList();
+        final List<String> updateUserGroups = updateRequest.getAllowedUserGroups();
+        if (updateName == null || updateHostList == null || updateUserGroups == null) {
+            throw new UnsupportedOperationException("Name, hostList and allowedUserGroups should be specified for "
+                    + "successful partition update");
+        }
+
+        final Context context = createContextWithUpdatePartitionName(updateName);
+        final CommandResult sinfoResult = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, context));
+        checkIfExecutionResultIsEmpty(sinfoResult);
+
+        final List<List<String>> partitionData = getNodesData(sinfoResult.getStdOut());
```

Comment:
maybe we could use some new class instead of `List<String>` to avoid `List<List<String>>` and non-obvious logic like `nodeData.get(some int)` on the next code rows

Reply:
fixed